File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/TextInputS.lua

module("ALittleIDE", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs
local ___coroutine = coroutine


assert(TextEditS, " extends class:TextEditS is nil")
TextInputS = ALittle.Class(TextEditS, "ALittleIDE.TextInputS")

function TextInputS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_textinput")
end

function TextInputS:LoadNatureBase()
	TextEditS.LoadNatureBase(self)
	self:LoadBoolData("password_mode", false, g_IDEEnum.yn_type)
	self:LoadDefaultNilString("default_text")
	self:LoadValueData("default_text_alpha")
	self:LoadValueData("ims_padding")
end

function TextInputS:HandlePasswordModeSELECT_CHANGE(event)
	self:BoolSelectChange("password_mode", false, g_IDEEnum.yn_type)
end

function TextInputS:HandleDefaultTextFOCUSOUT(event)
	self:DefaultNoStringInputChange("default_text", false)
end

function TextInputS:HandleDefaultTextAlphaFOCUSOUT(event)
	self:Z2OValueInputChange("default_text_alpha", false)
end

function TextInputS:HandleImsPaddingFOCUSOUT(event)
	self:ValueNumInputChange("ims_padding", false)
end

function TextInputS:HandleRegexFOCUSOUT(event)
	self:DefaultNoStringInputChange("regex", false)
end

function TextInputS:HandleLimitLenFOCUSOUT(event)
	self:ValueNumInputChange("limit_len", false)
end

File: ALittleIDE/Module/ALittleIDE/JSScript/IDE/UIEdit/Setting/TextInputS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.TextEditS === undefined) throw new Error(" extends class:ALittleIDE.TextEditS is undefined");
ALittleIDE.TextInputS = JavaScript.Class(ALittleIDE.TextEditS, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_textinput";
	},
	LoadNatureBase : function() {
		ALittleIDE.TextEditS.LoadNatureBase.call(this);
		this.LoadBoolData("password_mode", false, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadDefaultNilString("default_text");
		this.LoadValueData("default_text_alpha");
		this.LoadValueData("ims_padding");
	},
	HandlePasswordModeSELECT_CHANGE : function(event) {
		this.BoolSelectChange("password_mode", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleDefaultTextFOCUSOUT : function(event) {
		this.DefaultNoStringInputChange("default_text", false);
	},
	HandleDefaultTextAlphaFOCUSOUT : function(event) {
		this.Z2OValueInputChange("default_text_alpha", false);
	},
	HandleImsPaddingFOCUSOUT : function(event) {
		this.ValueNumInputChange("ims_padding", false);
	},
	HandleRegexFOCUSOUT : function(event) {
		this.DefaultNoStringInputChange("regex", false);
	},
	HandleLimitLenFOCUSOUT : function(event) {
		this.ValueNumInputChange("limit_len", false);
	},
}, "ALittleIDE.TextInputS");

}